src/core/Akka.Persistence/Snapshot/SnapshotStore.cs

```diff
@@ -75,15 +75,15 @@ private bool ReceiveSnapshotStore(object message)
             }
             else if (message is SaveSnapshot saveSnapshot)
             {
-                var metadata = new SnapshotMetadata(saveSnapshot.Metadata.PersistenceId, saveSnapshot.Metadata.SequenceNr, DateTime.UtcNow);
+                var metadata = new SnapshotMetadata(saveSnapshot.Metadata.PersistenceId, saveSnapshot.Metadata.SequenceNr, saveSnapshot.Metadata.Timestamp == DateTime.MinValue ? DateTime.UtcNow : saveSnapshot.Metadata.Timestamp);
```

Comment:
Fix, instead of always replacing the metadata timestamp with `DateTime.UtcNow`, we only replace it if it is not set, i.e. has a `TimeStamp.MinValue` value